tests/performance/timing_navier_stokes.cc

```diff
@@ -0,0 +1,2298 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+//
+// Description:
+//
+// A performance benchmark based on step 67 and step 76, extended to the
+// compressible Navier-Stokes equations. We measure timings for grid creation,
+// setup of unknowns, explicit Runge-Kutta time stepping with face-centric
+// loop through MatrixFree::loop() (step-67 style) as well as cell-centric
+// loop through MatrixFree::loop_cell_centric() (step-76 style). We use a
+// problem with periodic boundary conditions to avoid
+//
+// Status: experimental
```

Comment:
what does that mean? Maybe remove this statement?

Reply:
I'll let @tamiko comment on this one here, I took it from the other cases e.g. https://github.com/dealii/dealii/blob/d89583ef4d48771ff13caae26c21157ccd8dd96a/tests/performance/timing_step_37.cc#L23-L25 or https://github.com/dealii/dealii/blob/d89583ef4d48771ff13caae26c21157ccd8dd96a/tests/performance/timing_step_3.cc#L21-L23 - I guess we can adjust if we have settled with some cases.

The motivation with this PR is to add the relevant code with realistic operations stressing some of our core algorithms, while maintaining a reasonable balance between different parts. Though the balance is not as good as in the step-37 performance test, as the init of the mesh and dofs is much cheaper than the setup of `MatrixFree` (which has a bottleneck that should be addressed at some point in the future) or the 20 time steps with a high-order scheme.